fs/fs_zenfs.cc

```diff
@@ -786,6 +787,41 @@ IOStatus ZenFS::GetFileSize(const std::string& f, const IOOptions& options,
   return s;
 }
 
+IOStatus ZenFS::RollbackAuxDirRename(
+    const std::string& source_path, const std::string& dest_path,
+    const std::vector<std::string>& renamed_children, const IOOptions& options,
+    IODebugContext* dbg) {
+  IOStatus s;
+
+  for (const auto& rollback_child : renamed_children) {
+    std::string source_rollback_child_path =
+        (std::filesystem::path(dest_path) /
+         std::filesystem::path(rollback_child))
+            .string();
+    std::string dest_rollback_child_path =
+        (std::filesystem::path(source_path) /
+         std::filesystem::path(rollback_child))
+            .string();
+    s = RenameFileNoLock(source_rollback_child_path, dest_rollback_child_path,
+                         options, dbg);
+    if (!s.ok()) {
+      return IOStatus::Corruption(
+          "ZenFS RenameFileNoLock: Unable to roll back unsuccessful "
+          "ZenFS file renameing.");
+    }
+  }
+
+  s = target()->RenameFile(ToAuxPath(dest_path), ToAuxPath(source_path),
+                           options, dbg);
+  if (!s.ok()) {
+    return IOStatus::Corruption(
+        "ZenFS RenameFileNoLock: Unable to roll back auxiliary path "
+        "renameing after unsuccessful ZenFS file renameing.");
```

Comment:
Error msg feels a bit log, a simplified version will be helpful, I guess ? may be
`RenameFileNoLock: Failed to roll back auxiliary path renaming`